### ISSUE TITLE
Partial Keypaths

### ISSUE THREAD

**Issue by joshwils82** [ractivejs/ractive#2521]
I have a question on the keypath returned from Ractive.getNodeInfo.  I am creating an app using a bunch of dynamic partials each with its own context (typically a subnode in the root data).

When I inspect the keypath returned from Ractive.getNodeInfo it looks like:

`${${${${${page-content||{}}-body||{}}-rows-0-||{}}-content||{}}-sections-0-||{}}`

It appears it has all the information I would need to resolve it back to a node in the root data.

My root data looks like this:
```
{
  page: {
    content: {
      body: {
        rows: [
          {
            content: {
              sections: [
                {
                  text: {
                    text: 'Column 1'
				  }
				}
			  ]
			}
		  }
		]
	  }
	}
  }
}
```
I'm wondering if there is already a function or some way to resolve this keypath to an actual node in my data?

**Comment by evs-chris** [comment]
What version of Ractive? I'm guessing 0.7.3 based on the keypath, but it never hurts to ask.

I ask because context and aliasing are things addressed by edge, which will become 0.8. There are also new keypath helpers for sticking paths together. There is also an open PR for more helpful `getNodeInfo` in #2442.

It looks like you're using something like `{{>somePartial foo || {}}}`, which is equivalent to `{{#with foo || {}}}{{>somePartial}}{{/with}}` and gets you outside of globally synced model territory - because `foo || {}` is an expression it's no longer addressable as just data. Your best bet would probably be to handle the defaulting during during initialization with a pre-populated data object. If that's not possible, you can do something truly terrible like `{{~/makeSureThisIsInited(.foo, @keypath + '.foo')}}` where `makeSureThisIsInited` is something like `function(value, path) { if (!value) ractive.set(path, {}); return ''; }`. Then a subsequent `{{>somePartial .foo}}` would always have a value because the side-effecting 😱 interpolator just above it makes sure it is inited.

**Comment by joshwils82** [comment]
Yep 0.7.3, Thanks for the suggestions.  I will give your pre-populated data suggestion a go.  I'm looking forward to the new features in 0.8.  Great work!
